lib/kitchen/provisioner/chef_base.rb

```diff
@@ -133,6 +133,47 @@ class ChefBase < Base
 
       default_config :checksum
 
+      # Deprecate config attributes
+      deprecate_config_for :require_chef_omnibus do |provisioner|
+        if provisioner[:require_chef_omnibus] != defaults[:require_chef_omnibus]
```

Comment:
Note that this condition is just an example for how the blocks will return messages. Value of `false` will describe how to use the `install_strategy` attr and a version value will describe how to apply that value to the `product_version` attr.